cpp/ql/test/library-tests/dataflow/DefaultTaintTracking/defaulttainttracking.cpp

```diff
@@ -208,4 +208,50 @@ void test_field_to_obj_test_pointer_arith(Point* pp) {
   (pp + sizeof(*pp))->x = getenv("VAR")[0];
   sink(pp); // tainted [field -> object]
   sink(pp + sizeof(*pp)); // tainted [field -> object]
-}
\ No newline at end of file
+}
+
+void sink(char **);
+
+void test_pointers1()
+{
+	char buffer[1024];
+	char *s = getenv("VAR");
+	char *ptr1, **ptr2;
+	char *ptr3, **ptr4;
+
+	ptr1 = buffer;
+	ptr2 = &ptr1;
+	memcpy(buffer, s, 1024);
+	ptr3 = buffer;
+	ptr4 = &ptr3;
+
+	sink(buffer); // tainted
+	sink(ptr1); // tainted
+	sink(ptr2);
+	sink(*ptr2); // tainted [NOT DETECTED]
```

Comment:
I think this sink (and the one on line 234) aren't caught is due to the two-level indirection (i.e., `ptr2` is a `char**`, and not just a `char*`).